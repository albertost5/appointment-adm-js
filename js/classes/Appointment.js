class Appointment {
    constructor() {
        this.appointments = [];
    }

    newAppointment( appointment ) {
        this.appointments = [ ...this.appointments, appointment ];
    }

    replaceAppointment( appointment ) {
        const index = this.appointments.findIndex( appoinmnt => appoinmnt.id === appointment.id );
        this.appointments[ index ] = appointment;
    }
}

export default Appointment;

