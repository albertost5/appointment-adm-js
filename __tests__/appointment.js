import Appointment from "../js/classes/Appointment.js";


describe('Testing Appoinment class', () => {

    const appointment = new Appointment();
    const id = 1661616567676;

    it('1. Add new appoinment', () => {
        const appointmentObj = {
            brand: 'BMW',
            owner: 'Alberto',
            telephone: '1234',
            date: '11/11/2022',
            time: '11:11',
            description: 'New car!',
            id
        }

        appointment.newAppointment( appointmentObj );

        expect(appointment).toMatchSnapshot();
    });

    it('2. To delete an appoinment', () => {

        const newAppointment = {
            brand: 'BMW',
            owner: 'Alberto',
            telephone: '1234',
            date: '12/11/2022',
            time: '10:00',
            description: 'First revision!',
            id
        }

        appointment.replaceAppointment( newAppointment );
        
        expect(appointment).toMatchSnapshot();
    });

});