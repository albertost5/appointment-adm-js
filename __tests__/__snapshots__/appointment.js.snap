// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Appoinment class 1. Add new appoinment 1`] = `
Appointment {
  "appointments": [
    {
      "brand": "BMW",
      "date": "11/11/2022",
      "description": "New car!",
      "id": 1661616567676,
      "owner": "Alberto",
      "telephone": "1234",
      "time": "11:11",
    },
  ],
}
`;

exports[`Testing Appoinment class 2. To delete an appoinment 1`] = `
Appointment {
  "appointments": [
    {
      "brand": "BMW",
      "date": "12/11/2022",
      "description": "First revision!",
      "id": 1661616567676,
      "owner": "Alberto",
      "telephone": "1234",
      "time": "10:00",
    },
  ],
}
`;
